Legacy/bonej/src/main/java/org/bonej/plugins/ParticleDisplay.java

```diff
@@ -0,0 +1,477 @@
+
+	/** Surface colour style */
+	static final int GRADIENT = 0;
+	static final int SPLIT = 1;
+	static final int ORIENTATION = 2;
+
+	// ----------------- STACK DISPLAY ---------------//
+
+	/**
+	 * Create an image showing some particle measurement
+	 *
+	 * @param imp            an image.
+	 * @param particleLabels the particles in the image.
+	 * @param values         list of values whose array indices correspond to
+	 *                       particlelabels
+	 * @return ImagePlus with particle labels substituted with some value
+	 */
+	static ImagePlus displayParticleValues(final ImagePlus imp, final int[][] particleLabels, final double[] values) {
+		final int w = imp.getWidth();
+		final int h = imp.getHeight();
+		final int d = imp.getImageStackSize();
+		final int wh = w * h;
+		final float[][] pL = new float[d][wh];
+		values[0] = 0; // don't colour the background
+		final ImageStack stack = new ImageStack(w, h);
+		for (int z = 0; z < d; z++) {
+			for (int i = 0; i < wh; i++) {
+				final int p = particleLabels[z][i];
+				pL[z][i] = (float) values[p];
+			}
+			stack.addSlice(imp.getImageStack().getSliceLabel(z + 1), pL[z]);
+		}
+		final double max = Arrays.stream(values).max().orElse(0.0);
+		final ImagePlus impOut = new ImagePlus(imp.getShortTitle() + "_" + "volume", stack);
+		impOut.setCalibration(imp.getCalibration());
+		impOut.getProcessor().setMinAndMax(0, max);
+		return impOut;
+	}
+
+	/**
+	 * Display the particle labels as an ImagePlus
+	 *
+	 * @param particleLabels particles labelled in the original image.
+	 * @param imp            original image, used for image dimensions, calibration
+	 *                       and titles
+	 * @return an image of the particles.
+	 */
+	static ImagePlus displayParticleLabels(final int[][] particleLabels, final ImagePlus imp) {
+		final int w = imp.getWidth();
+		final int h = imp.getHeight();
+		final int d = imp.getImageStackSize();
+		final int wh = w * h;
+		final ImageStack stack = new ImageStack(w, h);
+		double max = 0;
+		for (int z = 0; z < d; z++) {
+			final float[] slicePixels = new float[wh];
+			for (int i = 0; i < wh; i++) {
+				slicePixels[i] = particleLabels[z][i];
+				max = Math.max(max, slicePixels[i]);
+			}
+			stack.addSlice(imp.getImageStack().getSliceLabel(z + 1), slicePixels);
+		}
+		final ImagePlus impParticles = new ImagePlus(imp.getShortTitle() + "_parts", stack);
+		impParticles.setCalibration(imp.getCalibration());
+		impParticles.getProcessor().setMinAndMax(0, max);
+		if (max > ConnectedComponents.MAX_LABEL)
+			IJ.error("Warning", "More than 2^23 particles. "
+					+ "Particle label display is imprecise above this number due to int to float conversion.");
+		return impParticles;
+	}
+
+	/**
+	 * 
+	 * @param imp
+	 * @param ellipsoids
+	 * @param title
+	 * @return ImagePlus containing particles drawn as best-fit solid ellipsoids
+	 */
+	static ImagePlus displayParticleEllipsoids(final ImagePlus imp, final Object[] ellipsoids, final String title) {
```

Comment:
`title` is always `"Ellipsoids"` so the parameter isn't really needed

Reply:
Fixed, good catch.